Build/test workflow fails with Gradle error

Started happening [yesterday](https://github.com/jspecify/jspecify-reference-checker/actions/runs/5050594865).

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':spotlessCheck'.
> Could not create task ':spotlessJavaCheck'.
   > Could not create task ':spotlessJava'.
      > File signature can only be created for existing regular files, given: /home/runner/work/jspecify-reference-checker/checker-framework/checker-qual/build/libs/checker-qual-3.21.5-SNAPSHOT.jar
```

--- a/build.gradle
+++ b/build.gradle
@@ -133,6 +133,29 @@ clean.doFirst {
     delete "${rootDir}/tests/build/"
 }
 
+/*
+ Spotless validates its formatters' dependencies eagerly, on project configuration.
+ google-java-format depends on checker-qual, which is built by a subproject.
+ On a clean build, the checker-qual JAR file doesn't exist yet, so Spotless throws an error
+ The file doesn't have to be correct; it just has to be a JAR file.
+ So here, before the spotless block,  we create a meaningless JAR file at that location if it doesn't already exist.
+ See https://github.com/jspecify/jspecify-reference-checker/issues/81
+ */
+
+def cfQualJar = file("${cfHome}/checker-qual/build/libs/checker-qual-${cfVersion}.jar")
+
+if (!cfQualJar.exists()) {
+    mkdir(cfQualJar.parent)
+    exec {
+        executable 'jar'
+        args = [
+            'cf',
+            cfQualJar.path,
+            buildFile.path // Use this build script file!
+        ]
+    }
+}
+
 spotless {
     java {
         googleJavaFormat()

--- a/build.gradle
+++ b/build.gradle
@@ -136,7 +136,7 @@ clean.doFirst {
 /*
  Spotless validates its formatters' dependencies eagerly, on project configuration.
  google-java-format depends on checker-qual, which is built by a subproject.
- On a clean build, the checker-qual JAR file doesn't exist yet, so Spotless throws an error
+ On a clean build, the checker-qual JAR file doesn't exist yet, so Spotless throws an error.
  The file doesn't have to be correct; it just has to be a JAR file.
  So here, before the spotless block,  we create a meaningless JAR file at that location if it doesn't already exist.
  See https://github.com/jspecify/jspecify-reference-checker/issues/81

